### PR TITLE
Update types for last version "draft-convert"

### PR DESCRIPTION
**Pull request by vadim-ch**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/HubSpot/draft-convert>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/draft-convert/index.d.ts [hunk 1]
```diff
@@ -1,73 +1,94 @@
...
+        type: T;
+    };
 
-    // Entity styling:
-    entityToHTML?: (entity: RawDraftEntity, originalText: string) => Tag;
-  }
+    type ExtendedHTMLElement<T> = (HTMLElement | HTMLLinkElement) & T;
 
-  type EntityKey = string;
+    interface IConvertToHTMLConfig<S, B, E> {
```
Comment (rbuckton) [inline types/draft-convert/index.d.ts:29]: Given this is a minor version bump, can you apply defaults for these type arguments so as not to break anyone who may have imported `IConvertToHTMLConfig` and used it without type arguments?

File: types/draft-convert/index.d.ts [hunk 2]
```diff
@@ -1,73 +1,94 @@
...
-    // Text entities
-    textToEntity?: (
-      text: string,
-      createEntity: (type: string, mutability: string, data: object) => EntityKey,
-      getEntity: (key: EntityKey) => Entity,
-      mergeEntityData: (key: string, data: object) => void,
-      replaceEntityData: (key: string, data: object) => void
-    ) => Array<{ entity: EntityKey; offset: number; length: number; result?: string }>;
-  }
+    interface IConvertFromHTMLConfig<S, B, E extends RawDraftEntity> {
```
Comment (rbuckton) [inline types/draft-convert/index.d.ts:42]: Just as with `IConvertFromHTMLConfig`, providing defaults for type arguments would be preferred.

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Review by orta — APPROVED** [review]
@vadim-ch & @MunifTanjim it would be good to get a review, but otherwise this seems reasonable to me

**Comment by vadim-ch** [comment]
Ready to merge

**Pull request merged**